### PR TITLE
Update the README testing example to one that works

### PR DESCRIPTION
**Pull request by michaelherold**
The example of how to testing Interactors in the README does not work in
the current version of `interactor`. This is a huge barrier to entry for
anyone looking to use the library. We should update the README to
indicate how tests should actually be done.

Note: This commit is just intended to add the spec to show that it
fails. I will add a second commit in a moment to show the method that
I use to test my Interactors.

### CODE REVIEW

**Comment by michaelherold** [comment]
Hopefully this PR is structured in a way that makes sense. I can add another commit that removes `spec/readme_spec.rb` if that's not something that should stick around. If that is wanted, I'll add a `[ci skip]` since it would just be a documentation update at that point.

Thanks for making such a fun interactor library. I'm really enjoying it compared to the previous library I used and am liking some of the discussion for the future of the gem.

**Comment by posgarou** [comment]
I was running into the same difficulty just now, looking at the tests in the README.

I think you can also modify the test in the README to replace calls of `interactor.call` with `interactor.run` and still get the expected behavior.  ([See the code for run here.](https://github.com/collectiveidea/interactor/blob/750ccad3003dfd4e6dbb1a5d1be472effc4dc1a1/lib/interactor.rb#L97-L117))

**Comment by michaelherold** [comment]
Hmm, fair enough. I liked the `#call` syntax better, as it feels more like issuing a command.

**Comment by philipgiuliani** [comment]
Looks good :) @posgarou the .run should not be used, its an internal method and may change in every update.

File: README.md [hunk 1]
```diff
@@ -521,18 +507,10 @@ describe AuthenticateUser do
...
-      end
+      it { is_expected.to be_a_failure }
 
       it "provides a failure message" do
-        expect {
-          interactor.call
-        }.to change {
-          context.message
-        }.from(nil).to be_present
+        expect(subject.message).to_not be_present
```
Comment (tbugai) [inline README.md]: If I'm reading this correctly the original expectation is that the messages goes from nil to being set after the subject has been invoked.  The new expectation asserts that message isn't set after the subject has been invoked.  That doesn't line up with the name of the test or what it is replacing.

**Comment by laserlemon** [comment]
I merged #84 with an updated readme because it applied cleanly. Thank you for your help!

**Pull request closed without merging**